typescript/libs/service_proxy.py

```diff
@@ -272,3 +272,40 @@ def create_req_dict(self, command_name, args=None):
         if args:
             req_dict["arguments"] = args
         return req_dict
+
+    def get_errors_in_file(self, path):
+        args = {
+            "file": path
+        }
+        req_dict = self.create_req_dict("semanticDiagnosticsSync", args)
+        json_str = json_helpers.encode(req_dict)
+        response_dict = self.__comm.sendCmdSync(json_str, req_dict["seq"])
+        return response_dict
+
+
+    def get_errors_at_cursor(self, path, cursor):
```

Comment:
I don't think `get_errors_at_cursor` should be in `service_proxy.py`. Also, to me it would make more sense to only request fixes at a position if the plugin has already reported a diagnostic on a given region (so you don't have to send a separate diagnostics request). In other words, I'm not sure requesting a code fix should request diagnostics.

Instead, maybe requesting a code fix at a position should check with the global `ProjectErrorListener` and see if it's been `modified`. If so, do nothing, but if not, filter through the errors as you've done here.

Curious to get @billti's thoughts here.

Reply:
After a bit of exploring 

Current diagnostics only contain `errorMessageString` and `region` leaving out `errorCodes` and `lineOffsets`(which are both required arguments to `getCodeFixes`). I tried passing along `errorCodes` and `offsets` but it didn't come out well with existing code structure.

Definitely agree there is a wasted call to server, but for the amount of refactor  and existing code base constraints it could be a bigger effort

I updated other things